### PR TITLE
Feature/df 1738 resolve bfs code

### PR DESCRIPTION
**Pull request by 99999989**

### CODE REVIEW

File: alv-portal-ui/src/app/shared/backend-services/reference-service/occupation-label.repository.ts [hunk 1]
```diff
@@ -19,6 +19,9 @@ export enum OccupationTypes {
   X28 = 'X28',
   SBN3 = 'SBN3',
   SBN5 = 'SBN5',
+  BFS = 'BFS',
+  CHISCO3 = 'CHISCO3',
```
Comment (99999989) [inline alv-portal-ui/src/app/shared/backend-services/reference-service/occupation-label.repository.ts]: @cjpsycos  Is CH-ISCO3 still needed

File: alv-portal-ui/src/app/shared/backend-services/reference-service/occupation-label.types.ts [hunk 2]
```diff
@@ -34,3 +34,10 @@ export interface OccupationCode {
   type: string;
   mapping?: { value: string; type: string };
 }
+
+export interface CatalogOccupationCode {
+  id?: string;
+  value: string;
+  type: string;
+  mappings?: { [key: string]: string };
```
Comment (99999989) [inline alv-portal-ui/src/app/shared/backend-services/reference-service/occupation-label.types.ts]: `mappings?: { [key: 'BFS' | 'CHISCO3']: string };`

File: alv-portal-ui/src/app/shared/occupations/occupation-suggestion.service.ts [hunk 3]
```diff
@@ -103,6 +108,16 @@ export class OccupationSuggestionService {
     return occupationCode;
   }
 
+  private toCatalogOccupationCode(occupation: OccupationLabelSuggestion) {
```
Comment (99999989) [inline alv-portal-ui/src/app/shared/occupations/occupation-suggestion.service.ts]: specify return type

File: alv-portal-ui/src/app/shared/occupations/occupation-suggestion.service.ts [hunk 4]
```diff
@@ -103,6 +108,16 @@ export class OccupationSuggestionService {
...
+    const occupationCode: CatalogOccupationCode = {
```
Comment (99999989) [inline alv-portal-ui/src/app/shared/occupations/occupation-suggestion.service.ts]: copy the object

File: alv-portal-ui/src/app/shared/occupations/occupation-suggestion.service.ts [hunk 5]
```diff
@@ -52,6 +53,10 @@ export class OccupationSuggestionService {
     return this.fetch(query, [OccupationTypes.AVAM], this.toJobPublicationOccupations);
   }
 
+  fetchCompetenceOccupations(query: string): Observable<Array<OccupationTypeaheadItem>> {
```
Comment (cjpsycos) [inline alv-portal-ui/src/app/shared/occupations/occupation-suggestion.service.ts]: fetchCompetenceCatalogOccupations

File: alv-portal-ui/src/app/shared/occupations/occupation-suggestion.service.ts [hunk 6]
```diff
@@ -10,6 +10,7 @@ import {
 } from './occupation-typeahead-item';
 import { map } from 'rxjs/operators';
 import {
+  CatalogOccupationCode,
```
Comment (cjpsycos) [inline alv-portal-ui/src/app/shared/occupations/occupation-suggestion.service.ts]: CompetenceCatalogOccupationCode

File: alv-portal-ui/src/app/shared/occupations/occupation-suggestion.service.ts (hunk 3 above)
Comment (cjpsycos) [inline alv-portal-ui/src/app/shared/occupations/occupation-suggestion.service.ts]: toCompetenceCatalogOccupationCode

File: alv-portal-ui/src/app/shared/backend-services/reference-service/occupation-label.repository.ts (hunk 1 above)
Comment (99999989) [reply alv-portal-ui/src/app/shared/backend-services/reference-service/occupation-label.repository.ts]: ok

**Pull request merged**